pandas/tests/frame/test_sorting.py

```diff
@@ -455,38 +455,38 @@ def test_sort_index_duplicates(self):
         df = DataFrame([lrange(5, 9), lrange(4)],
                        columns=['a', 'a', 'b', 'b'])
 
-        with tm.assert_raises_regex(ValueError, 'duplicate'):
+        with tm.assert_raises_regex(ValueError, 'not unique'):
             # use .sort_values #9816
             with tm.assert_produces_warning(FutureWarning):
                 df.sort_index(by='a')
-        with tm.assert_raises_regex(ValueError, 'duplicate'):
+        with tm.assert_raises_regex(ValueError, 'not unique'):
             df.sort_values(by='a')
 
-        with tm.assert_raises_regex(ValueError, 'duplicate'):
+        with tm.assert_raises_regex(ValueError, 'not unique'):
             # use .sort_values #9816
             with tm.assert_produces_warning(FutureWarning):
                 df.sort_index(by=['a'])
-        with tm.assert_raises_regex(ValueError, 'duplicate'):
+        with tm.assert_raises_regex(ValueError, 'not unique'):
             df.sort_values(by=['a'])
 
-        with tm.assert_raises_regex(ValueError, 'duplicate'):
+        with tm.assert_raises_regex(ValueError, 'not unique'):
             # use .sort_values #9816
             with tm.assert_produces_warning(FutureWarning):
                 # multi-column 'by' is separate codepath
                 df.sort_index(by=['a', 'b'])
-        with tm.assert_raises_regex(ValueError, 'duplicate'):
+        with tm.assert_raises_regex(ValueError, 'not unique'):
             # multi-column 'by' is separate codepath
             df.sort_values(by=['a', 'b'])
 
         # with multi-index
         # GH4370
         df = DataFrame(np.random.randn(4, 2),
                        columns=MultiIndex.from_tuples([('a', 0), ('a', 1)]))
-        with tm.assert_raises_regex(ValueError, 'levels'):
+        with tm.assert_raises_regex(ValueError, 'not unique'):
```

Comment:
I think for this case the old error message was more informative, so it would be nice if we could keep a separate message for the case of incomplete key for MI compared to just non-unique key

Reply:
Ok, sure. The new error message will read as follows (for the `MultiIndex` case):
```
ValueError: The column label 'a' is not unique.
For a multi-index, the label must be a tuple with elements corresponding to each level.
```
Does that sound clear? @jorisvandenbossche